data_pipeline/dataset.py

```diff
@@ -146,37 +156,47 @@ def __getitem__(self, item):
     id = self.ids[item]
     sentence = self.fields_by_id[id][SENTENCE_KEY]
     concepts = self.fields_by_id[id][CONCEPTS_KEY]
+    glove_concepts = self.fields_by_id[id][GLOVE_CONCEPTS_KEY]
     adj_mat = self.fields_by_id[id][ADJ_MAT_KEY]
     amr_str = self.fields_by_id[id][AMR_STR_KEY]
-    return id, sentence, concepts, adj_mat, amr_str
+    concepts_str = self.fields_by_id[id][CONCEPTS_STR_KEY]
+    return id, sentence, concepts, glove_concepts, adj_mat, amr_str, concepts_str
 
   def collate_fn(self, batch):
     amr_ids = []
     batch_sentences = []
     batch_concepts = []
+    batch_glove_concepts = []
     batch_adj_mats = []
     amr_strings = []
     sentence_lengths = []
     concepts_lengths = []
+    concepts_str = []
     for entry in batch:
-      amr_id, sentence, concepts, adj_mat, amr_str = entry
+      amr_id, sentence, concepts, glove_concepts, adj_mat, amr_str, concepts_as_str = entry
       amr_ids.append(amr_id)
       batch_sentences.append(sentence)
       batch_concepts.append(concepts)
+      batch_glove_concepts.append(glove_concepts)
       batch_adj_mats.append(adj_mat)
       amr_strings.append(amr_str)
+      concepts_str.append(concepts_as_str)
       sentence_lengths.append(len(sentence))
       concepts_lengths.append(len(concepts))
     # Get max lengths for padding.
     max_sen_len = max([len(s) for s in batch_sentences])
     max_concepts_len = max([len(s) for s in batch_concepts])
+    max_glove_concepts_len = max([len(s) for s in batch_glove_concepts])
     max_adj_mat_size = max([len(s) for s in batch_adj_mats])
     # Pad sentences.
     padded_sentences = [
       torch_pad(s, (0, max_sen_len - len(s))) for s in batch_sentences]
     # Pad concepts
     padded_concepts = [
       torch_pad(c, (0, max_concepts_len - len(c))) for c in batch_concepts]
+    # Pad glove concepts
+    padded_glove_concepts = [
+      torch_pad(c, (0, max_glove_concepts_len - len(c))) for c in batch_glove_concepts]
```

Comment:
Just a question, shouldn't we also pad the concepts_str?

Reply:
No, because I use this for displaying the adj mat image on tensorboard, to compare the relations between each concept. I don't need the padding there.
Think about it this way, for 3 concepts and a fake root we would have a 4x4 matrix, very visual. But for example for bolt, for one of the batches the max_seq_len is 17, why would I tranform that matrix in such a big one, and I dont even care about the padding, and neither does the model when training, because it's ignored when applying padding mask.
I would add the padding here only yo get rid of it later..